packages/next/src/state.tsx

```diff
@@ -0,0 +1,153 @@
+"use client";
+
+export type Maybe<TType> = TType | null;
+import React, {
+  type PropsWithChildren,
+  createContext,
+  useContext,
+  useRef,
+} from "react";
+import { createStore, useStore } from "zustand";
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface SfLocales {}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface SfCart {}
```

Comment:
Types cannot be taken from UDL package as those can be extended on the middleware layer. Using generics was also not possible due to the fact that `createContext` cannot be wrapped with a function and it's impossible to pass generics to consts. It's created in line 52 of this file.

Reply:
Not applicable anymore. It has been changed.